GVFS/GVFS.Common/Maintenance/PrefetchStep.cs

```diff
@@ -291,6 +297,53 @@ private bool TrySchedulePostFetchJob(List<string> packIndexes)
             return false;
         }
 
+        /// <summary>
+        /// Ensure the prefetch pack with most-recent timestamp has an associated
+        /// ".keep" file. This prevents any Git command from deleting the pack.
+        ///
+        /// Delete the previous ".keep" file so that pack can be deleted when they
+        /// are not the most-recent pack.
+        /// </summary>
+        private void UpdateKeepPacks()

```

Comment:
Can we add a test for this new functionality.  Would it be an easy extention maybe of existing Prefetch functional tests?